### PR TITLE
Store pickup storefront improvement

### PR DESCRIPTION
**Pull request by swnsma**
### Description (*)
1. If geolocation data is missed, load Pickup Locations without area filter
Steps:
- select offline distance calculation algorithm (have not tested with Google);
- create few sources in the USA, Texas and upgrade them to pickup locations;
- go to checkout as a guest;
- select In-Store Pickup shipping method;
- open Pickup Location selection popup;
**E.R.**
- some of available pickup locations will be shown in the pickup locations list..
**A.R.**
- no pickup locations will be shown in the pickup locations list.

2. Add runtime cache for frontend application.
Same search term will sent to the web-server on each request.

### CODE REVIEW

**Comment by novikor** [comment]
@magento run Static Tests

**Review by novikor — DISMISSED** [review]
Please fix static tests:
https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/inventory/pull/3052/b39ea81fef20b02738537bb80ccc64f5/Statics/allure-report-b2b/index.html#suites/3a29d43e5c1686787ff8c2a77e590abb/fa924bfe579e169e/

File: InventoryInStorePickupFrontend/view/frontend/web/js/view/store-selector.js [hunk 1]
```diff
@@ -118,21 +117,33 @@ define([
...
         },
 
+        /**
+         * Get Search Term from search query and country.
+         *
+         * @param {String} searchQuery
+         * @param {String} country
+         * @returns {String}
+         */
+        getSearchTerm: function(searchQuery, country) {
```
Comment (novikor) [inline InventoryInStorePickupFrontend/view/frontend/web/js/view/store-selector.js]:
```suggestion
        getSearchTerm: function (searchQuery, country) {
```

**Comment by novikor** [comment]
@magento run Static Tests

**Comment by novikor** [comment]
@magento run Static Tests

**Comment by novikor** [comment]
@magento run Static Tests

**Comment by sidolov** [comment]
@magento run all tests

**Comment by sidolov** [comment]
@magento run Static Tests

**Comment by sidolov** [comment]
@magento run WebAPI Tests,Unit Tests,Semantic Version Checker,Magento Health Index,Integration Tests,Functional Tests EE,Functional Tests CE,Functional Tests B2B

**Review by sidolov — APPROVED** [review]

**Comment by sidolov** [comment]
@magento run all tests

**Comment by sidolov** [comment]
@magento run WebAPI Tests

**Comment by sidolov** [comment]
@magento run all tests

**Pull request merged**